R/dummy_multi_choice.R

```diff
@@ -0,0 +1,307 @@
+#'  Multi Choice Dummy Variables Creation
+#'
+#' `step_dummy_multi_choice()` creates a *specification* of a recipe
+#'  step that will convert multiple nominal data (e.g. character or factors)
+#'  into one or more numeric binary model terms for the levels of
+#'  the original data.
+#'
+#' @inheritParams step_center
+#' @inherit step_center return
+#' @param ... One or more selector functions to choose which variables will be
+#'  used to compute the components. See [selections()] for more details.
+#' @param role For model terms created by this step, what analysis
+#'  role should they be assigned?. By default, the function assumes
+#'  that the binary dummy variable columns created by the original
+#'  variables will be used as predictors in a model.
+#' @param threshold A numeric value between 0 and 1. If it's less than one then
+#'  factor levels whose rate of occurrence in the training set are below
+#'  `threshold` will be "othered". If it is 0 then no thresholding will be done.
+#' @param levels A list that contains the information needed to
+#'  create dummy variables for each variable contained in
+#'  `terms`. This is `NULL` until the step is trained by
+#'  [prep.recipe()].
+#' @param input A character vector containing the names of the columns used.
+#'  This is `NULL` until the step is trained by [prep.recipe()].
+#' @param other A single character value for the "other" category.
+#' @param naming A function that defines the naming convention for
+#'  new dummy columns. See Details below.
```

Comment:
Would be a good idea to `@inheritParams` or `@template` but I feel it should be done in a separate PR since it would hit multiple other steps. `naming` is used in `step_dummy()` and `step_ratio()`

Reply:
I ended up using `@inheritParams step_dummy`